src/compiler/datetime.rs

```diff
@@ -84,3 +84,12 @@ impl TryFrom<String> for TimeZone {
         }
     }
 }
+
+impl From<TimeZone> for FixedOffset {
```

Comment:
also see my [comment here](https://github.com/vectordotdev/vector/pull/18506/files#r1344284369)

Reply:
Thanks @pront, I've implemented the conversion to remove duplication in the Vector repo.